plugins/filter/json_patch.py

```diff
@@ -0,0 +1,176 @@
+    HAS_LIB = False
+    JSONPATCH_IMPORT_ERROR = exc
+else:
+    HAS_LIB = True
+    JSONPATCH_IMPORT_ERROR = None
+
+OPERATIONS_AVAILABLE = ["add", "copy", "move", "remove", "replace", "test"]
+OPERATIONS_NEEDING_FROM = ["copy", "move"]
+OPERATIONS_NEEDING_VALUE = ["add", "replace", "test"]
+
+
+class FilterModule:
+    """Filter plugin."""
+
+    def check_json_object(
+        self, filter_name: str, object_name: str, inp: Union[str, list, dict]
+    ):
+        if not isinstance(inp, (str, list, dict)):
+            raise AnsibleFilterError(
+                f"{filter_name}: {object_name} is not dictionary, list or string"
+            )
+
+    def check_patch_arguments(self, filter_name: str, args: dict):
+
+        if "op" not in args or not isinstance(args["op"], str):
+            raise AnsibleFilterError(f"{filter_name}: 'op' argument is not a string")
+
+        if args["op"] not in OPERATIONS_AVAILABLE:
+            raise AnsibleFilterError(
+                f"{filter_name}: unsupported 'op' argument: {args['op']}"
+            )
+
+        if "path" not in args or not isinstance(args["path"], str):
+            raise AnsibleFilterError(f"{filter_name}: 'path' argument is not a string")
+
+        if args["op"] in OPERATIONS_NEEDING_FROM:
+            if "from" not in args:
+                raise AnsibleFilterError(
+                    f"{filter_name}: 'from' argument missing for '{args['op']}' operation"
+                )
+            if not isinstance(args["from"], str):
+                raise AnsibleFilterError(
+                    f"{filter_name}: 'from' argument is not a string"
+                )
+
+    def json_patch(
+        self,
+        inp: Union[str, list, dict],
+        op: str,
+        path: str,
+        value: Any = None,
+        **kwargs: dict,
+    ) -> Any:
+
+        if not HAS_LIB:
+            raise AnsibleFilterError(
+                "You need to install 'jsonpatch' package prior to running 'json_patch' filter"
+            ) from JSONPATCH_IMPORT_ERROR
+
+        self.check_json_object("json_patch", "input", inp)
+
+        args = {"op": op, "path": path}
+        from_arg = kwargs.pop("from", None)
+
+        if kwargs:
+            raise AnsibleFilterError(
+                f"json_patch: unexpected keywords arguments: {', '.join(kwargs.keys())}"
+            )
+
+        if op in OPERATIONS_NEEDING_VALUE:
+            args["value"] = value
+        if op in OPERATIONS_NEEDING_FROM and from_arg is not None:
+            args["from"] = from_arg
+
+        self.check_patch_arguments("json_patch", args)
+
+        result = None
+
+        try:
+            result = jsonpatch.apply_patch(
+                inp if isinstance(inp, (dict, list)) else loads(inp),
+                [args],
+            )
+        except jsonpatch.JsonPatchTestFailed:
+            pass
+        except Exception as e:
+            raise AnsibleFilterError(f"json_patch: patch failed: {to_native(e)}") from e
```

Comment:
`to_native` is no longer necessary for plugins (since they no longer support Python 2.7). You can simply use `str()` for this use-case, or (even better) simply leave it away since f-strings automatically call `str()` on non-strings:
```suggestion
            raise AnsibleFilterError(f"json_patch: patch failed: {e}") from e
```